answerking_app/tests/test_unit/test_unit_productserializer.py

```diff
@@ -0,0 +1,211 @@
+from decimal import Decimal
+import copy
+
+from rest_framework.utils.serializer_helpers import ReturnDict
+
+from answerking_app.models.models import Category, Product
+from answerking_app.models.serializers import ProductSerializer
+from answerking_app.tests.test_unit.UnitTestBaseClass import UnitTestBase
+
+
+class ProductSerializerTests(UnitTestBase):
+
+    utb = UnitTestBase()
+
+    serializer_path: str = "answerking_app.models.serializers."
+    test_cat_data: dict = utb.get_fixture(
+        "categories", "burgers_cat_data.json"
+    )
+    test_prod_data: dict = utb.get_fixture(
+        "products", "plain_burger_data.json"
+    )
+    test_product_serializer: ProductSerializer = ProductSerializer(
+        test_prod_data
+    )
+
+    def setUp(self):
+        cat: Category = Category.objects.create(**self.test_cat_data)
+        prod: Product = Product.objects.create(**self.test_prod_data)
+        cat.products.add(prod)
+
+    def tearDown(self):
+        Category.objects.all().delete()
+        Product.objects.all().delete()
+
+    def test_product_serializer_contains_correct_fields(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializer(test_prod).data
+        expected: list[str] = [
+            "id",
+            "name",
+            "description",
+            "price",
+            "categories",
+            "retired",
+        ]
+        actual: list[str] = list(test_serializer_data.keys())
+        self.assertCountEqual(actual, expected)
+
+    def test_product_serializer_id_field_content(self):
```

Comment:
This was suggested by the serializer testing template. I think it separates each field into different tests so that if something goes wrong with one field we know immediately which field it is rather than having to investigate.